charts/metrics-server/values.yaml

```
@@ -130,3 +130,18 @@ tolerations: []
 affinity: {}
 
 topologySpreadConstraints: []
+
+tls:
+  enable: false
+  ca: |-
+    # Public CA file that signed the APIService
+  key: |-
+    # Private key of the APIService
+  certificate: |-
+    # Public key of the APIService
+
+certManager:
+  enabled: false
+  caCertDuration: 43800h
+  certDuration: 8760h
```

```suggestion
tls:
  secret: ""

  certManager:
    enabled: false
    caCertDuration: 43800h
    certDuration: 8760h
```

How about the following inputs? TLS is enabled if `tls.secret` is set or if `tls.certManager.enabled` is `true`. If both are set cert-manager should use the secret name. You can then define `metrics-server.tlsSecretName` in __helpers.tpl_ to return `tls.secret` if set or generate a name based on the full name with a `-tls-cert` suffix if not which can be used in the Certificate and Deployment resources.